packages/nouns-contracts/test/foundry/rewards/ProposalRewards.t.sol

```diff
@@ -294,6 +294,53 @@ contract ProposalRewardsTest is BaseProposalRewardsTest {
         assertEq(rewards.clientBalance(clientId2), 0.15 ether); // 15 eth * 1%
     }
 
+    function test_onlyEligibleProposalsCanSetTheRewardsPeriod() public {
+        uint256 startTimestamp = block.timestamp;
+
+        bidAndSettleAuction({ bidAmount: 5 ether });
+        bidAndSettleAuction({ bidAmount: 10 ether });
+
+        vm.warp(startTimestamp + 2 weeks + 1);
+        proposeVoteAndEndVotingPeriod(clientId2);
+
+        settleAuction();
+        bidAndSettleAuction({ bidAmount: 100 ether });
+
+        // trying to create a bogus proposal to capture the high auction as reward
+        uint32 proposalId = uint32(propose(bidder2, address(1), 1 ether, '', '', 'my proposal', 0));
+        mineBlocks(VOTING_DELAY + UPDATABLE_PERIOD_BLOCKS + VOTING_PERIOD + 1);
+
+        votingClientIds = [0];
+        rewards.updateRewardsForProposalWritingAndVoting({
+            lastProposalId: proposalId,
+            votingClientIds: votingClientIds
+        });
+
+        assertEq(rewards.clientBalance(clientId2), 0.15 ether); // 15 eth * 1%
+    }
+
+    function test_cantUseIneligibleProposalToPassTheMinimumPeriod() public {
```

Comment:
maybe worth further explaining with comments the state you're creating here?

* we do not pass `numProposalsEnoughForReward`, hence it'll revert
* then we have the time filter which looks at the creation timestamp of the most recent eligible proposal - and that's the constraint an attacker might try to circumvent, and that's what we're showing won't work here

Reply:
added: 2669f0174a3524aabbed5da5d804534f1fe54de7